middlewares/osio/auth.go

```diff
@@ -168,10 +166,16 @@ func (a *OSIOAuth) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.
 				return
 			}
 
+			isSerivce, err := a.CheckSrvAccToken(token)
+			if err != nil {
+				log.Errorf("Invalid token, %v", err)
+				rw.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+
 			var cached cacheData
-			service := getService(r)
-			if service == che {
-				userID := r.Header.Get(impersonate)
+			if isSerivce {
+				userID := r.URL.Query().Get("identity_id")
```

Comment:
Now using "identity_id" from request param.  I guess che team aware of this integration point.